### DISCUSSION TITLE
Behavior of API endpoint with multiple filter

### DISCUSSION THREAD

**Discussion by Flocon29** [netbox-community/netbox#7543] [Q&A]
Hello,

I have a small question about the API:

I am using this request : ```/api/extras/config-contexts/?name=config1&&name=config2```
With this, I have the expected behavior, I receive both config-context : `config1` & `config2`

Now when I am using this request : ```/api/extras/custom-fields/?name=custom1&&name=custom2```
I don't have the behavior that I need, I receive only the last element `custom2` and not both.

However I read it from the documentation : https://netbox.readthedocs.io/en/stable/rest-api/filtering/
```However, a logical AND operation will be used in instances where a field may have multiple values, such as tags```

But here, for custom-fields, the field name can't have multiple values, can it?
Is this a bug, a normal behavior or am I doing it wrong?

Thanks in advance for your help.

**Comment by candlerb** [comment]
(Aside: you should use `&` to separate the URL query string parameters, not `&&`.  This is just the structure of URLs, it doesn't mean "and")

In cases where a field cannot have multiple values, then repeating the query normally acts as a logical "OR".  For example:

```
curl -g 'https://xxxxxx/api/dcim/devices/?name=nuc1&name=nuc2'
```

will return devices where the name is nuc1 *or* nuc2.

However, I've tested it and you're correct that this doesn't work for `/api/extras/custom-fields/`.  Only the last matching one is returned.  I think it would be reasonable to open a bug report for this.

(EDIT: it's the same for `name=foo&name=bar` or `id=1&id=2`. Only the custom field named `bar` or id `2` is returned)

**Reply by Flocon29** [reply]
Hi @candlerb, thanks for your efficiency.

Since we agree on the default behavior on `custom-fields` request, I will immediately open a bug report about it!

(EDIT : And yes, about `&` or `&&` pattern, I test both, and the behavior is the same on ```/api/extras/config-contexts``` request)
